tests/e2e/lib/virt_helpers.go

```diff
@@ -0,0 +1,408 @@
+	namespace := "openshift-cnv"
+
+	csv, version, err := GetCsvFromPackageManifest(dynamicClient, "kubevirt-hyperconverged")
+	if err != nil {
+		log.Printf("Failed to get CSV from package manifest")
+		return nil, err
+	}
+
+	v := &VirtOperator{
+		Client:    client,
+		Clientset: clientset,
+		Dynamic:   dynamicClient,
+		Namespace: namespace,
+		Csv:       csv,
+		Version:   version,
+	}
+
+	return v, nil
+}
+
+// GetCsvFromPackageManifest returns the current CSV from the first channel
+// in the given PackageManifest name. Uses the dynamic client because adding
+// the real PackageManifest API from OLM was actually more work than this.
+func GetCsvFromPackageManifest(dynamicClient dynamic.Interface, name string) (string, *version.Version, error) {
+	resourceId := schema.GroupVersionResource{
+		Group:    "packages.operators.coreos.com",
+		Resource: "packagemanifests",
+		Version:  "v1",
+	}
+
+	log.Println("Getting packagemanifest...")
+	unstructuredManifest, err := dynamicClient.Resource(resourceId).Namespace("default").Get(context.Background(), name, v1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting packagemanifest %s: %v", name, err)
+		return "", nil, err
+	}
+
+	log.Println("Extracting channels...")
+	channels, ok, err := unstructured.NestedSlice(unstructuredManifest.UnstructuredContent(), "status", "channels")
+	if err != nil {
+		log.Printf("Error getting channels from packagemanifest: %v", err)
+		return "", nil, err
+	}
+	if !ok {
+		return "", nil, errors.New("failed to get channels list from " + name + " packagemanifest")
+	}
+	if len(channels) < 1 {
+		return "", nil, errors.New("no channels listed in package manifest " + name)
+	}
+
+	firstChannel, ok := channels[0].(map[string]interface{})
+	if !ok {
+		return "", nil, errors.New("failed to read first channel from package manifest " + name)
+	}
+
+	csv, ok, err := unstructured.NestedString(firstChannel, "currentCSV")
+	if err != nil {
+		return "", nil, err
+	}
+	if !ok {
+		return "", nil, errors.New("failed to get current CSV from " + name + " packagemanifest")
+	}
+	log.Printf("Current CSV is: %s", csv)
+
+	versionString, ok, err := unstructured.NestedString(firstChannel, "currentCSVDesc", "version")
+	if err != nil {
+		return "", nil, err
+	}
+	if !ok {
+		return "", nil, errors.New("failed to get current operator version from " + name + " packagemanifest")
+	}
+	log.Printf("Current operator version is: %s", versionString)
+
+	version, err := version.ParseGeneric(versionString)
+	if err != nil {
+		return "", nil, err
+	}
+
+	return csv, version, nil
+}
+
+// Checks the existence of the operator's target namespace
+func (v *VirtOperator) checkNamespace() bool {
+	// First check that the namespace exists
+	exists, _ := DoesNamespaceExist(v.Clientset, v.Namespace)
+	return exists
+}
+
+// Checks for the existence of the virtualization operator group
+func (v *VirtOperator) checkOperatorGroup() bool {
```

Comment:
These ones (check*, install*, ensure*) are definitely not meant to be called from outside. GetOperatorSubscription is capitalized just to match the existing subscription helper code, but lower case does make more sense because it is only used privately.

The others that are still capitalized need to stay that way so they can be called from the test suite code.